src/Symfony/Component/Form/FormFactory.php

```diff
@@ -126,7 +126,7 @@ public function createBuilderForProperty($class, $property, $data = null, array
 
         // user options may override guessed options
         if ($typeGuess) {
-            $options = array_merge($typeGuess->getOptions(), $options);
+            $options = array_replace_recursive($typeGuess->getOptions(), $options);
```

Comment:
> Losing the maxlength attribute when passing the attr option is expected IMO [...]

It's not a BC break? before this PR it's not true and the issue description proposes merge `attr` options also for guessedOptions and populated options, so now the one that is lost is me :)